lib/async-git-prompt.zsh

```diff
@@ -0,0 +1,33 @@
+ASYNC_PROC=0
+function precmd() {
+    function async() {
+        # save to temp file
+        printf "%s" "$(git_prompt_info)" > "${HOME}/.zsh_tmp_prompt"
+
+        # signal parent
+        kill -s USR1 $$
+    }
+
+    # do not clear RPROMPT, let it persist
+
+    # kill child if necessary
+    if [[ "${ASYNC_PROC}" != 0 ]]; then
+        kill -s HUP $ASYNC_PROC >/dev/null 2>&1 || :
```

Comment:
Can use `&>/dev/null` instead of `>/dev/null 2>&1` if you want to be more concise.